modules/post/linux/dos/xen_420_dos.rb

```diff
@@ -225,16 +228,16 @@ def test_success
   ##
 
   def make_code
-    m = <<-END
-obj-m := #{@c_name}.o
+    m = <<~END
```

Comment:
is this always a safe change to make? I _think_ it's fine here but doesn't this alter the whitespace?

Reply:
> is this always a safe change to make? I _think_ it's fine here but doesn't this alter the whitespace?

Not always, but in this instance it is fine.

The Makefile tab formatting is preserved with `\t`. Preceding whitespace is ignored when using `~END` (as opposed to `-END`), so long as all lines are correctly/equally indented.

```
# cat asdf.rb 
@c_name = 'asdf'

def make_code
    m = <<~END
      obj-m := #{@c_name}.o
      EXTRA_CFLAGS+= -save-temps
      all:
      \t$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
      clean:
      \t$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
    END
    m
  end

puts make_code

# ruby asdf.rb 
obj-m := asdf.o
EXTRA_CFLAGS+= -save-temps
all:
	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
clean:
	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
```